bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckablePullRequest.java

```diff
@@ -202,25 +204,30 @@ Hash amendManualReviewers(Hash commit, Namespace namespace, Hash original) throw
         }
     }
 
-    PullRequestCheckIssueVisitor createVisitor(Hash hash) throws IOException {
-        var checks = JCheck.checksFor(localRepo, hash);
+    PullRequestCheckIssueVisitor createVisitor(JCheckConfiguration conf, Hash hash) throws IOException {
+        var checks = JCheck.checksFor(conf, localRepo, hash);
         return new PullRequestCheckIssueVisitor(checks);
     }
 
-    void executeChecks(Hash localHash, CensusInstance censusInstance, PullRequestCheckIssueVisitor visitor,
-                       List<String> additionalConfiguration, Hash hash) throws IOException {
-        Optional<JCheckConfiguration> conf;
-        if (confOverride != null) {
-            conf = JCheck.parseConfiguration(confOverride, additionalConfiguration);
-        } else {
-            conf = JCheck.parseConfiguration(localRepo, hash, additionalConfiguration);
-        }
-        if (conf.isEmpty()) {
-            throw new RuntimeException("Failed to parse jcheck configuration at: " + hash + " with extra: " + additionalConfiguration);
+    Optional<JCheckConfiguration> parseJCheckConfiguration(Hash hash) throws IOException {
```

Comment:
Every caller of this method is throwing IllegalStateException if jcheck conf isn't found. Did you consider just throwing here and not return Optional? I realize there is a bit more information in the exception message, but on the other hand, whoever reads it will have a stacktrace anyway.